### PR TITLE
emphasize update() in developer manual

### PR DESCRIPTION
**Pull request by Paul-St-Young**
## Proposed changes

Emphasize the need to call `ParticleSet::update()` in the developer manual.
The coexistence of `R` and `RSoA` did not sink in for me until now.
I spent many hours trying to figure out why my unit test distance tables are garbage...
Just want to add some emphasis in the manual to help the next unfortunate soul.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

workstation

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

File: docs/developing.rst [hunk 1]
```diff
@@ -780,6 +780,29 @@ needs to be transferred to the structure-of-arrays (SoA) storage in
...
+  const SimulationCell sc;
+  ParticleSet elec(sc), ions(sc);
+  elec.setName("e");
+  ions.setName("ion0");
+
+  // initialize ions
+  ions.create(2);
+  ions.R[0] = {0.0, 0.0, 0.0};
+  ions.R[1] = {0.5, 0.5, 0.5};
+  ions.setCoordinates(ions.R); // transfer to RSoA
```
Comment (Paul-St-Young) [inline docs/developing.rst:795]: I didn't realize that this line was needed.
Without it, the distance table contains garbage.

Comment (ye-luo) [reply docs/developing.rst:795]: Should avoid calling ``P.setCoordinates(P.R)``. Calling an public API with an internal data is bad. The documentation was clear, calling "P.update()" is necessary. Doesn't it work?

Comment (Paul-St-Young) [reply docs/developing.rst:795]: `ions.update()` does work here.
I saw the `setCoordinates` example in `test_polynomial_eeI_jastrow.cpp` so I thought it was OK.

File: docs/developing.rst [hunk 2]
```diff
@@ -780,6 +780,30 @@ needs to be transferred to the structure-of-arrays (SoA) storage in
...
+  ParticleSet elec(sc), ions(sc);
+  elec.setName("e");
+  ions.setName("ion0");
+
+  // initialize ions
+  ions.create(2);
+  ions.R[0] = {0.0, 0.0, 0.0};
+  ions.R[1] = {0.5, 0.5, 0.5};
+  ions.update(); // transfer to RSoA
+  //ions.setCoordinates(ions.R); // avoid public API with internal data
```
Comment (ye-luo) [inline docs/developing.rst:796]: Remove this line.

File: docs/developing.rst [hunk 3]
```diff
@@ -780,6 +780,30 @@ needs to be transferred to the structure-of-arrays (SoA) storage in
...
+
+  // initialize elec
+  elec.create(2);
+  elec.R[0] = {0.0, 0.0, 0.0};
+  elec.R[1] = {0.0, 0.25, 0.0};
+  const int itab = elec.addTable(ions);
+  elec.update(); // update RSoA and distance tables
+
+  const auto& d_table = elec.getDistTableAB(itab);
+  // d_table is an electron-ion distance table
```
Comment (ye-luo) [inline docs/developing.rst:806]: Move this comment line above by one line.

**Review by ye-luo — CHANGES_REQUESTED** [review]
The manual rendering doesn't seem right. Please review the manual build in CI.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**